lib/manageiq/loggers/journald.rb

```diff
@@ -28,12 +44,59 @@ def add(severity, message = nil, progname = nil)
         end
 
         message = formatter.call(format_severity(severity), progname, message)
+        caller_object = caller_locations.last
 
-        Systemd::Journal.print(log_level_map[severity], message)
+        # TODO: Allow the syslog facility to be set via the configuration settings.
+        Systemd::Journal.message(
+          :message           => message,
+          :message_id        => message_id,
```

Comment:
I think message_id is specific to the log-line rather than the logger instance, http://0pointer.de/blog/projects/journal-submit.html

> MESSAGE_ID= is a 128bit ID that identifies our specific message call, formatted as hexadecimal string. We randomly generated this string with journalctl --new-id128. This can be used by applications to track down all occasions of this specific message. The 128bit can be a UUID, but this is not a requirement or enforced.

Reply:
Right, I was trying to find a way we might be able to use that field. Since we don't have any real canned messages that we could tag, do you have another suggestion? Or should I just drop it?

Reply:
Lets not set it by default, we can allow callers to set it per-message in the future

Reply:
Alright, removed it.